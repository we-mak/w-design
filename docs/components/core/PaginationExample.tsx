import { Pagination } from "@w-design/core";

const PaginationExample = ({ defaultSelectedIndex }) => {
  return (
    <Pagination
      pages={[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]}
      defaultSelectedIndex={defaultSelectedIndex}
      onChange={(page: number) => {
        console.log(page);
      }}
    />
  );
};

export default PaginationExample;
