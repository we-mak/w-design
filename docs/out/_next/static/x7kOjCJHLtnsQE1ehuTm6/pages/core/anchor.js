(window.webpackJsonp=window.webpackJsonp||[]).push([[15],{"E/Ix":function(t,e,n){"use strict";n.d(e,"a",(function(){return c})),n.d(e,"b",(function(){return p}));var a=n("mXGw"),r=n.n(a),l=r.a.createContext({}),b=function(t){var e=r.a.useContext(l),n=e;return t&&(n="function"===typeof t?t(e):Object.assign({},e,t)),n},c=function(t){var e=b(t.components);return r.a.createElement(l.Provider,{value:e},t.children)};var i="mdxType",o={inlineCode:"code",wrapper:function(t){var e=t.children;return r.a.createElement(r.a.Fragment,{},e)}},u=Object(a.forwardRef)((function(t,e){var n=t.components,a=t.mdxType,l=t.originalType,c=t.parentName,i=function(t,e){var n={};for(var a in t)Object.prototype.hasOwnProperty.call(t,a)&&-1===e.indexOf(a)&&(n[a]=t[a]);return n}(t,["components","mdxType","originalType","parentName"]),u=b(n),p=a,m=u[c+"."+p]||u[p]||o[p]||l;return n?r.a.createElement(m,Object.assign({},{ref:e},i,{components:n})):r.a.createElement(m,Object.assign({},{ref:e},i))}));function p(t,e){var n=arguments,a=e&&e.mdxType;if("string"===typeof t||a){var l=n.length,b=new Array(l);b[0]=u;var c={};for(var o in e)hasOwnProperty.call(e,o)&&(c[o]=e[o]);c.originalType=t,c[i]="string"===typeof t?t:a,b[1]=c;for(var p=2;p<l;p++)b[p]=n[p];return r.a.createElement.apply(null,b)}return r.a.createElement.apply(null,n)}u.displayName="MDXCreateElement"},Ff2n:function(t,e,n){"use strict";function a(t,e){if(null==t)return{};var n,a,r=function(t,e){if(null==t)return{};var n,a,r={},l=Object.keys(t);for(a=0;a<l.length;a++)n=l[a],e.indexOf(n)>=0||(r[n]=t[n]);return r}(t,e);if(Object.getOwnPropertySymbols){var l=Object.getOwnPropertySymbols(t);for(a=0;a<l.length;a++)n=l[a],e.indexOf(n)>=0||Object.prototype.propertyIsEnumerable.call(t,n)&&(r[n]=t[n])}return r}n.d(e,"a",(function(){return a}))},NPsZ:function(t,e,n){"use strict";n.r(e),n.d(e,"default",(function(){return p}));var a=n("wx14"),r=n("Ff2n"),l=n("mXGw"),b=n.n(l),c=n("E/Ix"),i=n("b0Hy"),o=(b.a.createElement,{}),u="wrapper";function p(t){var e=t.components,n=Object(r.a)(t,["components"]);return Object(c.b)(u,Object(a.a)({},o,n,{components:e,mdxType:"MDXLayout"}),Object(c.b)("h1",null,"AnchorScroll"),Object(c.b)("p",null,"Smooth scrolling when click to an anchor link in the page. This component is helpful for improve the user\ninteractive on the layout."),Object(c.b)("h2",null,"Example"),Object(c.b)(i.AnchorScroll,{destination:"target",callback:function(){return console.log("fire callback")},mdxType:"AnchorScroll"},Object(c.b)("span",{role:"img","aria-label":"img","aria-labelledby":"anchor link"},"\ud83d\ude80"),"To target"),Object(c.b)(i.AnchorScroll,{destination:500,mdxType:"AnchorScroll"},"Scroll to 500px"),Object(c.b)("pre",null,Object(c.b)("code",Object(a.a)({parentName:"pre"},{className:"language-jsx"}),'<AnchorScroll destination="target" callback={() => console.log("fire callback")}>\n  <span role="img" aria-label="img" aria-labelledby="anchor link">\ud83d\ude80</span>\n  To target\n</AnchorScroll>\n\n// And the target at somewhere\n<div id="target" style={{margin: "2rem 0"}}>\n  <span role="img" aria-label="img" aria-labelledby="anchor target">\ud83d\udcb0</span>\n  Target\n</div>\n')),Object(c.b)("h2",null,"Custom animation"),Object(c.b)("p",null,"AnchorScroll intially only use liner animation but you can pick one of below to set whatever kind you want"),Object(c.b)("pre",null,Object(c.b)("code",Object(a.a)({parentName:"pre"},{className:"language-js"}),"{\n  easeInQuad: (t: number) => t * t,\n  easeOutQuad(t: number) {\n    return t * (2 - t);\n  }\n  easeInOutQuad(t: number) {\n    return t < 0.5 ? 2 * t * t : -1 + (4 - 2 * t) * t;\n  }\n  easeInCubic(t: number) {\n    return t * t * t;\n  }\n  easeOutCubic(t: number) {\n    return --t * t * t + 1;\n  }\n  easeInOutCubic(t: number) {\n    return t < 0.5 ? 4 * t * t * t : (t - 1) * (2 * t - 2) * (2 * t - 2) + 1;\n  }\n  easeInQuart(t: number) {\n    return t * t * t * t;\n  }\n  easeOutQuart(t: number) {\n    return 1 - --t * t * t * t;\n  }\n  easeInOutQuart(t: number) {\n    return t < 0.5 ? 8 * t * t * t * t : 1 - 8 * --t * t * t * t;\n  }\n  easeInQuint(t: number) {\n    return t * t * t * t * t;\n  }\n  easeOutQuint(t: number) {\n    return 1 + --t * t * t * t * t;\n  }\n  easeInOutQuint(t: number) {\n    return t < 0.5 ? 16 * t * t * t * t * t : 1 + 16 * --t * t * t * t * t;\n  }\n}\n")),Object(c.b)("div",{id:"target",style:{margin:"2rem 0"}},Object(c.b)("span",{role:"img","aria-label":"img","aria-labelledby":"anchor target"},"\ud83d\udcb0"),"Target"),Object(c.b)("h2",null,"API"),Object(c.b)("table",null,Object(c.b)("thead",{parentName:"table"},Object(c.b)("tr",{parentName:"thead"},Object(c.b)("th",Object(a.a)({parentName:"tr"},{align:null}),"Props"),Object(c.b)("th",Object(a.a)({parentName:"tr"},{align:null}),"Type"),Object(c.b)("th",Object(a.a)({parentName:"tr"},{align:null}),"default"),Object(c.b)("th",Object(a.a)({parentName:"tr"},{align:null}),"description"))),Object(c.b)("tbody",{parentName:"table"},Object(c.b)("tr",{parentName:"tbody"},Object(c.b)("td",Object(a.a)({parentName:"tr"},{align:null}),"destination"),Object(c.b)("td",Object(a.a)({parentName:"tr"},{align:null}),"string, number"),Object(c.b)("td",Object(a.a)({parentName:"tr"},{align:null}),Object(c.b)("inlineCode",{parentName:"td"},"md")),Object(c.b)("td",Object(a.a)({parentName:"tr"},{align:null}),"target id or position cordinate")),Object(c.b)("tr",{parentName:"tbody"},Object(c.b)("td",Object(a.a)({parentName:"tr"},{align:null}),"duration"),Object(c.b)("td",Object(a.a)({parentName:"tr"},{align:null}),"number"),Object(c.b)("td",Object(a.a)({parentName:"tr"},{align:null}),Object(c.b)("inlineCode",{parentName:"td"},"200")),Object(c.b)("td",Object(a.a)({parentName:"tr"},{align:null}),"scroll duration time")),Object(c.b)("tr",{parentName:"tbody"},Object(c.b)("td",Object(a.a)({parentName:"tr"},{align:null}),"easing"),Object(c.b)("td",Object(a.a)({parentName:"tr"},{align:null}),"Function"),Object(c.b)("td",Object(a.a)({parentName:"tr"},{align:null}),Object(c.b)("inlineCode",{parentName:"td"},"(t: number) => t")),Object(c.b)("td",Object(a.a)({parentName:"tr"},{align:null}),"easing animation function, is linear by default")),Object(c.b)("tr",{parentName:"tbody"},Object(c.b)("td",Object(a.a)({parentName:"tr"},{align:null}),"margin"),Object(c.b)("td",Object(a.a)({parentName:"tr"},{align:null}),"number"),Object(c.b)("td",Object(a.a)({parentName:"tr"},{align:null}),Object(c.b)("inlineCode",{parentName:"td"},"0")),Object(c.b)("td",Object(a.a)({parentName:"tr"},{align:null}),"addition or subtraction scroll target point")),Object(c.b)("tr",{parentName:"tbody"},Object(c.b)("td",Object(a.a)({parentName:"tr"},{align:null}),"top"),Object(c.b)("td",Object(a.a)({parentName:"tr"},{align:null}),"number"),Object(c.b)("td",Object(a.a)({parentName:"tr"},{align:null}),Object(c.b)("inlineCode",{parentName:"td"},"0")),Object(c.b)("td",Object(a.a)({parentName:"tr"},{align:null}),"addition height, useful when has fixed menu on top")),Object(c.b)("tr",{parentName:"tbody"},Object(c.b)("td",Object(a.a)({parentName:"tr"},{align:null}),"callback"),Object(c.b)("td",Object(a.a)({parentName:"tr"},{align:null}),"Function"),Object(c.b)("td",Object(a.a)({parentName:"tr"},{align:null}),"undefined"),Object(c.b)("td",Object(a.a)({parentName:"tr"},{align:null}),"callback function after scrolled")))))}p.isMDXComponent=!0},T25V:function(t,e,n){(window.__NEXT_P=window.__NEXT_P||[]).push(["/core/anchor",function(){return n("NPsZ")}])},wx14:function(t,e,n){"use strict";function a(){return(a=Object.assign||function(t){for(var e=1;e<arguments.length;e++){var n=arguments[e];for(var a in n)Object.prototype.hasOwnProperty.call(n,a)&&(t[a]=n[a])}return t}).apply(this,arguments)}n.d(e,"a",(function(){return a}))}},[["T25V",0,1,2]]]);