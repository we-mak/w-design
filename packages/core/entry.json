{
  "index": "src/index.ts",
  "AnchorScroll": "src/AnchorScroll/index.tsx",
  "Avatar": "src/Avatar/index.tsx",
  "AvatarGroup": "src/AvatarGroup/index.tsx",
  "Box": "src/Box/index.ts",
  "Breadcrumbs": "src/Breadcrumbs/index.tsx",
  "Button": "src/Button/index.tsx",
  "Card": "src/Card/index.tsx",
  "Comment": "src/Comment/index.tsx",
  "Container": "src/Container/index.tsx",
  "CustomTag": "src/CustomTag/index.tsx",
  "DisplayElement": "src/DisplayElement/index.tsx",
  "Divider": "src/Divider/index.ts",
  "Drawer": "src/Drawer/index.tsx",
  "Dropdown": "src/Dropdown/index.tsx",
  "EmptyView": "src/EmptyView/index.tsx",
  "Flexbox": "src/Flexbox/index.tsx",
  "Icon": "src/Icon/index.tsx",
  "Image": "src/Image/index.tsx",
  "List": "src/List/index.tsx",
  "Menu": "src/Menu/index.tsx",
  "Modal": "src/Modal/index.tsx",
  "Navbar": "src/Navbar/index.tsx",
  "Picture": "src/Picture/index.tsx",
  "Portal": "src/Portal/index.tsx",
  "ProgressLoader": "src/ProgressLoader/index.tsx",
  "Provider": "src/Provider/index.tsx",
  "PushMessage": "src/PushMessage/index.tsx",
  "SectionMessage": "src/SectionMessage/index.tsx",
  "Spinner": "src/Spinner/index.tsx",
  "Typo": "src/Typo/index.tsx"
}
