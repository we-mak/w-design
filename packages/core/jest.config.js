const base = require("../../jest.config.base.js");
const pack = require("./package.json");

module.exports = {
  ...base,
  name: pack.name,
  displayName: pack.name,
  rootDir: "../..",
  testMatch: [
    `<rootDir>/packages/core/**/*.test.+(ts|tsx|js)`,
    `<rootDir>/packages/core/**/__tests__/*.+(ts|tsx|js)`
  ]
};
