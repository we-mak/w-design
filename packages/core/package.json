{
  "name": "@w-design/core",
  "version": "0.41.0-alpha.28+d990095",
  "description": "w-design core components",
  "keywords": [
    "w-design",
    "react",
    "core",
    "components"
  ],
  "author": "Son_dang <son.dang@doda100.com >",
  "homepage": "https://github.com/we-mak/w-design#readme",
  "license": "MIT",
  "main": "lib/cjs/index.js",
  "module": "lib/es/index.es.js",
  "types": "lib/cjs/index.d.ts",
  "sideEffects": false,
  "publishConfig": {
    "access": "public"
  },
  "directories": {
    "lib": "lib",
    "test": "__tests__"
  },
  "files": [
    "lib/cjs/*",
    "lib/es/*"
  ],
  "repository": {
    "type": "git",
    "url": "git+https://viiiprock@github.com/we-mak/w-design.git"
  },
  "scripts": {
    "test": "cross-env CI=1 jest test -u --colors"
  },
  "bugs": {
    "url": "https://github.com/we-mak/w-design/issues"
  },
  "dependencies": {
    "@types/body-scroll-lock": "^2.6.1",
    "@w-design/helpers": "^0.40.0",
    "@w-design/hooks": "^0.40.0",
    "body-scroll-lock": "^2.6.4"
  },
  "gitHead": "d990095e2f75e0d6e34aff66a98969a4de8eae6f"
}
