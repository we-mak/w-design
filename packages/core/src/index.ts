export { default as AnchorScroll } from "./AnchorScroll";
export { default as Avatar } from "./Avatar";
export { default as AvatarGroup } from "./AvatarGroup";
export { default as Badge } from "./Badge";
export { default as Box } from "./Box";
export { default as Breadcrumbs, BreadcrumbsItem } from "./Breadcrumbs";
export { default as Button, ButtonGroup } from "./Button";
export {
  default as Card,
  CardMedia,
  CardHeader,
  CardContent,
  CardFooter,
} from "./Card";

export * from "./Chip";
export { default as Chip } from "./Chip";

export { default as Collapse } from "./Collapse";
export { default as Comment } from "./Comment";
export { default as Container } from "./Container";
export { default as CustomTag } from "./CustomTag";
export { default as DisplayElement } from "./DisplayElement";
export { default as Divider } from "./Divider";
export { default as Drawer } from "./Drawer";
export { default as Dropdown } from "./Dropdown";
export { default as EmptyView } from "./EmptyView";
export { default as Flexbox } from "./Flexbox";
export { default as Icon } from "./Icon";
export { default as Image, ImageRatio } from "./Image";
export { default as List, ListItem, ListFooter } from "./List";
export { default as Menu, MenuItem, MenuHeading, SubMenu } from "./Menu";
export { default as Modal } from "./Modal";
export { default as Navbar } from "./Navbar";

export * from "./Pagination";
export { default as Pagination } from "./Pagination";

export { default as Picture } from "./Picture";
export { default as Portal } from "./Portal";

export * from "./Popover";
export { default as Popover } from "./Popover";

export { default as ProgressLoader } from "./ProgressLoader";
export { default as Provider } from "./Provider";
export { default as SectionMessage } from "./SectionMessage";

export * from "./Spinner";
export { default as Spinner } from "./Spinner";

export * from "./Steps";
export { default as Steps } from "./Steps";

export { default as Tabs, Tab } from "./Tabs";
export { default as Typo } from "./Typo";

export * from "./PushMessage";
export { default as PushMessage } from "./PushMessage";
