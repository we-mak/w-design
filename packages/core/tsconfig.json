{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "outDir": "lib",
    "declarationDir": "lib",
    "baseUrl": "."
  },
  "include": ["src/**/*"]
}
