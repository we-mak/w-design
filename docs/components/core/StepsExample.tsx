import { useState } from "react";
import { Steps, Button, ButtonGroup } from "@w-design/core";

export const StepsExample = () => {
  const [activeStepIndex, setActiveStepIndex] = useState(0);

  const prevStep = () => setActiveStepIndex(activeStepIndex - 1);
  const nextStep = () => setActiveStepIndex(activeStepIndex + 1);

  return (
    <Steps
      steps={[
        {
          title: "Step 1",
          content: <div>This is step 1</div>,
        },
        {
          title: "Step 2",
          content: <div>This is step 2</div>,
        },
        {
          title: "Step 3",
          content: <div>This is step 3</div>,
        },
      ]}
      activeIndex={activeStepIndex}
      controlGroup={
        <ButtonGroup>
          <Button onClick={prevStep}>Previous</Button>
          <Button onClick={nextStep}>Next</Button>
        </ButtonGroup>
      }
    />
  );
};
