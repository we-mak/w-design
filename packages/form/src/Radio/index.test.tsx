import * as React from "react";
import { render, cleanup, fireEvent } from "@testing-library/react";
import { Provider } from "@w-design/core";
import RadioGroup from "./index";

afterEach(cleanup);

const options = [
  {
    label: "Option 1",
    value: "foo",
    name: "option 1",
    isChecked: true
  },
  {
    label: "Option 2",
    value: "bar",
    isDisabled: true
  }
];

describe("<Radio/>", () => {
  // Render test
  it("should renders correctly", () => {
    const { container } = render(
      <Provider>
        <RadioGroup
          isRequired
          groupName="Radio"
          groupLabel="Radio"
          options={options}
        />
        <RadioGroup isDisabled defaultValue="foo" options={options} />
        <RadioGroup value="test" />
      </Provider>
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  // Event
  it("should fire onchange event", () => {
    const { getByLabelText } = render(
      <Provider>
        <RadioGroup
          groupName="Radio"
          groupLabel="Radio"
          options={[
            {
              label: "Foo",
              value: "foo"
            },
            {
              label: "Bar",
              value: "bar"
            }
          ]}
        />
      </Provider>
    );

    const radio: any = getByLabelText(/Foo/i);
    fireEvent.change(radio, { target: { value: "bar" } });

    expect(radio.value).toBe("bar");
  });

  it("should fire onRadioChange event", () => {
    const { getByLabelText } = render(
      <Provider>
        <RadioGroup
          groupName="Radio"
          groupLabel="Group"
          options={[
            {
              label: "Foo",
              value: "foo"
            },
            {
              label: "Bar",
              value: "bar"
            }
          ]}
        />
      </Provider>
    );

    fireEvent.click(getByLabelText(/Foo/i));
    expect((getByLabelText(/Foo/i) as any).checked).toBe(true);
  });

  it("should call onChange when it is a function", () => {
    const onChange = jest.fn();
    const { getByLabelText } = render(
      <Provider>
        <RadioGroup
          groupName="Radio"
          groupLabel="Group"
          options={[
            {
              label: "Foo",
              value: "foo"
            },
            {
              label: "Bar",
              value: "bar"
            }
          ]}
          onChange={onChange}
        />
      </Provider>
    );

    fireEvent.click(getByLabelText(/Foo/i));

    expect(onChange).toHaveBeenCalled();
  });
});
