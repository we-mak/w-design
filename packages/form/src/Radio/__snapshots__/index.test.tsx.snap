// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Radio/> should renders correctly 1`] = `
<div>
  <label
    class="sc-AykKG iRgEwA"
  >
    Radio
  </label>
  <label
    class="sc-AykKD hcJZZv"
  >
    <input
      aria-label="Option 1"
      class="sc-AykKF OkLcO"
      name="option 1"
      required=""
      type="radio"
      value="foo"
    />
    <div
      class="sc-AykKC cHdQRz"
    >
      <svg
        focusable="false"
        height="28"
        role="presentation"
        viewBox="0 0 24 24"
        width="28"
      >
        <g
          fill-rule="evenodd"
        >
          <circle
            cx="12"
            cy="12"
            fill="currentColor"
            r="7"
          />
          <circle
            cx="12"
            cy="12"
            fill="inherit"
            r="3"
          />
        </g>
      </svg>
    </div>
    <div
      class="sc-AykKE hhOwrg"
    >
      Option 1
    </div>
  </label>
  <label
    class="sc-AykKD hcJZZv"
  >
    <input
      aria-label="Option 2"
      class="sc-AykKF OkLcO"
      disabled=""
      name="Radio"
      required=""
      type="radio"
      value="bar"
    />
    <div
      class="sc-AykKC cHdQRz"
    >
      <svg
        focusable="false"
        height="28"
        role="presentation"
        viewBox="0 0 24 24"
        width="28"
      >
        <g
          fill-rule="evenodd"
        >
          <circle
            cx="12"
            cy="12"
            fill="currentColor"
            r="7"
          />
          <circle
            cx="12"
            cy="12"
            fill="inherit"
            r="3"
          />
        </g>
      </svg>
    </div>
    <div
      class="sc-AykKE hhOwrg"
    >
      Option 2
    </div>
  </label>
</div>
`;
